src/main/java/de/zalando/aruha/nakadi/repository/db/EventTypeCache.java

```diff
@@ -0,0 +1,140 @@
+            } else {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public void created(EventType eventType) {
+        try {
+            String path = getZNodePath(eventType.getName());
+            zkClient
+                    .create()
+                    .creatingParentsIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath(path, new byte[0]);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void removed(String name) {
+        try {
+            String path = ZKPaths.makePath(ZKNODE_PATH, name);
+            zkClient.delete().forPath(path);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void initParentCacheZNode(CuratorFramework zkClient) throws Exception {
+        try {
+            zkClient
+                    .create()
+                    .creatingParentsIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath(ZKNODE_PATH);
+        } catch (KeeperException.NodeExistsException e) {
+            System.out.println("NODE ALREADY EXISTS");
+        }
+    }
+
+    private void addCacheChangeListener(final LoadingCache<String, EventType> cache, PathChildrenCache cacheSync) {
+        PathChildrenCacheListener listener = new PathChildrenCacheListener() {
+            @Override
+            public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) throws Exception {
+                switch ( event.getType() ) {
+                    case CHILD_UPDATED: {
+                        invalidateCacheKey(event);
+                        break;
+                    }
+                    case CHILD_REMOVED: {
+                        invalidateCacheKey(event);
+                        break;
+                    }
+                }
+            }
+
+            private void invalidateCacheKey(PathChildrenCacheEvent event) {
+                String path[] = event.getData().getPath().split("/");
+
+                cache.invalidate(path[path.length - 1]);
+            }
+        };
+
+        cacheSync.getListenable().addListener(listener);
+    }
+
+    private PathChildrenCache setupCacheSync(CuratorFramework zkClient) throws Exception {
+        PathChildrenCache cacheSync = new PathChildrenCache(zkClient, ZKNODE_PATH, false);
+
+        cacheSync.start();
+
+        addCacheChangeListener(cache, cacheSync);
+
+        return cacheSync;
+    }
+
+    private LoadingCache<String,EventType> setupInMemoryCache(EventTypeRepository dbRepo) {
+        CacheLoader<String, EventType> loader = new CacheLoader<String, EventType>() {
+            public EventType load(String key) throws NoSuchEventTypeException {
+                return dbRepo.findByName(key);
+            }
+        };
+
+        return CacheBuilder.newBuilder().maximumSize(100000).build(loader);
```

Comment:
This is arbitrarily chosen number, just to make sure memory would not increase wildly. But I haven't checked for heap size if it would fit.